### PR TITLE
Feature/hadas/steps

### PR DESCRIPTION
**Pull request by hadasfa**
Description:
1. New steps component
2. fix focus state for on primary button
3. fix disable for on primary button

https://user-images.githubusercontent.com/72390374/127994760-3a70b421-78d7-42f7-8ecc-a1fb9677f23c.mov

https://user-images.githubusercontent.com/72390374/127994844-126864f1-1d64-4a7a-bffa-a4a4d13bda26.mov

"On primary without background color" - what do you think? (this is not happening with normal tertiary)

https://user-images.githubusercontent.com/72390374/127995327-051613f7-d51e-45fa-a68c-09104a0f229f.mov

#### Basic
- [V ] Used plop (npm run plop) to create a new component. This command will create a folder for your component and will update index.js file
- [V] PR has description
- [ V] New component is functional and using React Hooks. 
- [ ] New React Hooks were added to /src/hooks folder (optional)
- [ V] Component defines PropTypes
- [ ] There is Readme for the component, which explains the main aspects of how to use it
#### Style
- [ V] Styles are added to NewComponent.scss file inside of the NewComponent folder
- [ V] CSS selectors are named using [BEM convention](http://getbem.com/naming/) 
- [ V] Design is compatible with [Monday Design System](https://design.monday.com/)
#### Storybook
- [ V] Stories were added to /src/NewComponent/__stories__/NewComponent.stories.js file
- [ V] Stories include sandbox story with knobs for each property
- [ V] Stories include all flows of using the component
- [ V] Component passed Accessibility Plugin checks
#### Tests
- [V ] Tests are compliant with TESTING_README.md instructions

### CODE REVIEW

File: src/components/Steps/Steps.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React, { useRef, forwardRef } from "react";
```
Comment (hadasfa) [inline src/components/Steps/Steps.jsx:1]: should it be in the future under MultiStepIndicator or separated component?

File: src/components/Steps/StepsDot.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+export const StepsDot = ({
+  stepsCount,
+  onChangeActiveStep,
+  activeStepIndex,
+  stepIndex,
+  ariaCurrent,
+  stepDescriptionFunc
+}) => {
+  const isActive = activeStepIndex === stepIndex;
```
Comment (orrgottlieb) [inline src/components/Steps/StepsDot.jsx:20]: please pass it in the props there is no need to rerender all on every activeStepIndex

like in the parent component pass - `isActive={activeStepIndex === stepIndex}`

File: src/components/Steps/StepsGalleryHeader.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { StepsDot } from "./StepsDot";
+import { STEPS_CSS_BASE_CLASS } from "./StepsConstants";
+
+const CSS_BASE_CLASS = `${STEPS_CSS_BASE_CLASS}-header_gallery`;
+
+export const StepsGalleryHeader = ({
+  activeStepIndex,
+  stepsCount,
+  onChangeActiveStep,
+  groupAriaLabel,
```
Comment (orrgottlieb) [inline src/components/Steps/StepsGalleryHeader.js:11]: remove groupAriaLabel not in use

File: src/components/Steps/StepsGalleryHeader.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  stepDescriptionFunc
+}) => {
+  const defaultStepDescriptionFunc = useCallback(stepIndex => `Step number ${stepIndex}`, []);
+  const galleryDots = useMemo(() => {
+    const dots = [];
+    for (let stepIndex = 0; stepIndex < stepsCount; stepIndex++) {
```
Comment (orrgottlieb) [inline src/components/Steps/StepsGalleryHeader.js:17]: why not map?

File: src/components/Steps/StepsGalleryHeader.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      dots.push(
+        <StepsDot
+          key={`monday-style-step-dot-${stepIndex + 1}`}
```
Comment (orrgottlieb) [inline src/components/Steps/StepsGalleryHeader.js:20]: this is not unique, please check if we can have uniqueness here?

File: src/components/Steps/hooks/useChangeStepFunction.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { useCallback } from "react";
+
+export function useChangeStepFunction({ onClickCallback, activeStepIndex, stepsCount, newStepIndex }) {
+  debugger;
+  return useCallback(
+    e => {
+      debugger;
```
Comment (orrgottlieb) [inline src/components/Steps/hooks/useChangeStepFunction.js:7]: remove debugger

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

File: src/components/Steps/StepsGalleryHeader.js (hunk 5 above)
Comment (hadasfa) [reply src/components/Steps/StepsGalleryHeader.js:14]: i understand it is a bad practice but every solution i find is a bit hard  for this case(I dont know  what is the value of each step content - if the the value is a div or a react component or a string)

File: src/components/Steps/__stories__/steps.stories.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+import Steps from "../Steps";
+import { STEPS_GALLERY_TYPE, STEPS_NUMBERS_TYPE } from "../StepsConstants";
+import StoryWrapper from "../../../StoryBookComponents/StoryWrapper/StoryWrapper";
+import StoryTitle from "../../storybook-helpers/story-title/story-title";
+import "./steps.stories.scss";
+
+export const States = () => {
+  const [numbersIndex, setNumberIndex] = useState(0);
+  const changeIndex = (e, active) => {
+    // eslint-disable-next-line no-debugger
```
Comment (hadasfa) [inline src/components/Steps/__stories__/steps.stories.js:14]: will be removed

**Pull request merged**